runtime/shared_common/OSCache.hpp

```diff
@@ -61,7 +61,7 @@
 #define OSCACHE_LOWEST_ACTIVE_GEN 1
 
 /* Always increment this value by 2. For testing we use the (current generation - 1) and expect the cache contents to be compatible. */
-#define OSCACHE_CURRENT_CACHE_GEN 35
+#define OSCACHE_CURRENT_CACHE_GEN 37
```

Comment:
We need a different generation number when J9VM_OPT_VALHALLA_NESTMATES is defined, since the ROMClass format is different.

These functions also need to be modified to understand the new generation number. Its been suggested to change them to use if statements rather than a switch so they don't need to be changed again in the future.

SH_OSCache::getHeaderFieldOffsetForGen
SH_OSCacheFile::getMmapHeaderFieldOffsetForGen
SH_OSCachesysv::getSysvHeaderFieldOffsetForGen